aten/src/ATen/native/cuda/SortUtils.cuh

```diff
@@ -151,4 +153,101 @@ bitonicSortKVInPlace(at::cuda::detail::TensorInfo<K, IndexType> keys,
   }
 }
 
+template <int KeyDims, int ValueDims,
+          int block_size, int items_per_thread,
+          typename K, typename V, typename IndexType>
+C10_LAUNCH_BOUNDS_1(block_size)
+__global__ void
+radixSortKVInPlace(at::cuda::detail::TensorInfo<K, IndexType> keys,
```

Comment:
is this better than DeviceSegmentedRadix sort that's used for some configurations already? It seems like it should be pretty similar.

Reply:
Yes, it is significantly better. `DeviceSegmentedRadixSort` is used by `launch_stable_sort_kernel` and if you look at the stable sort speedup graph you can see `radixSortKVInPlace` is at worst 1.5x faster and at best 25x faster, depending length of the dimension being sorted.

Reply:
Hm, interesting, seems like cub should have gotten it right